ambry-messageformat/src/main/java/com.github.ambry.messageformat/CompositeBlobInfo.java

```diff
@@ -62,4 +126,49 @@ public int getChunkSize() {
   public List<StoreKey> getKeys() {
```

Comment:
An alternative to materializing a list of `StoreKey`s for every `CompositeBlobInfo` is to return a "view" of `chunkMetadataList`:
```java
  public List<StoreKey> getKeys() {
    return new AbstractList<StoreKey>() {
      @Override
      public StoreKey get(int index) {
        return chunkMetadataList.get(index).getStoreKey();
      }

      @Override
      public int size() {
        return chunkMetadataList.size();
      }
    };
  }
```

It seems like this would suffice for the two use cases that need the `getKeys()` method, the `getChunkIdsOnly` option in the router and `BlobIdTransformer`, as both of them are just interested in iterating through all of the keys and not mutating the list.

Reply:
Cool idea, I'll use it